templates/boltzmann.js

```diff
@@ -1070,6 +1179,61 @@ function _requireOr (target, value) {
   }
 }
 
+let _cookie = null
+class Cookie extends Map {
+  constructor(values) {
+    super(values)
+    this.changed = new Set()
+  }
+
+  set (key, value) {
+    if (this.changed) {
```

Comment:
So! `Map`'s constructor is defined as taking an iterable of tuples, and _then calling `this.set(t[0], t[1])` for each tuple_. This is great, because we want to massage the shape of items coming through. HOWEVER, this means we're calling it before `this.changed` is set! The tricky part is that you _can't modify `this`_ before calling `super()` in a child class.

I just thought of _maybe_ a way around it – class instance fields, which I still regard as "very new" – which I'll test, but that's the reason for this goofy `if` check.